pyro/infer/importance.py

```diff
@@ -140,3 +144,72 @@ def _fn(*args, **kwargs):
     if normalized:
         log_weights = log_weights - torch.logsumexp(log_weights)
     return log_weights, model_trace, guide_trace
+
+
+def psis_diagnostic(model, guide, *args, **kwargs):
```

Comment:
Is this function meaningfully differentiable? If you are certain it is not, then I'd recommend decorating with `@torch.no_grad()`:
```py
@torch.no_grad()
def psis_diagnostic(...):
    ...
```

Reply:
done